nats-base-client/kv.ts

```diff
@@ -158,24 +156,35 @@ export class Bucket implements KV, KvRemove {
   bucket: string;
   codec!: KvCodecs;
   _prefixLen: number;
+  subjPrefix: string;
 
   constructor(bucket: string, jsm: JetStreamManager, js: JetStreamClient) {
     validateBucket(bucket);
     this.jsm = jsm;
     this.js = js;
     this.bucket = bucket;
     this._prefixLen = 0;
+    this.subjPrefix = kvSubjectPrefix;
+
+    const jsi = js as JetStreamClientImpl;
+    const prefix = jsi.prefix || "$JS.API";
+    if (prefix !== "$JS.API") {
+      this.subjPrefix = `${prefix}.${kvSubjectPrefix}`;
+    }
   }
 
   static async create(
-    nc: NatsConnection,
+    js: JetStreamClient,
```

Comment:
I am a bit confused between passing in the JetStreamClient and creating the jetstream manager locally. 
Would passing in the options or the manager by any better?

Reply:
that is just internal - the clients get it as nc.jetstream().views.kv(name)

Reply:
ok